sigar_linux_test.go

```diff
@@ -280,4 +280,69 @@ DirectMap2M:      333824 kB
 			Expect(swap.Free).To(BeNumerically("==", 786428*1024))
 		})
 	})
+
+	Describe("DiskIO", func() {
+		var diskstatFile string
+		BeforeEach(func() {
+			diskstatFile = procd + "/diskstats"
+			diskstatContents := `
+   1       0 ram0 0 0 0 0 0 0 0 0 0 0 0
+   1       1 ram1 0 0 0 0 0 0 0 0 0 0 0
+   1       2 ram2 0 0 0 0 0 0 0 0 0 0 0
+   1       3 ram3 0 0 0 0 0 0 0 0 0 0 0
+   1       4 ram4 0 0 0 0 0 0 0 0 0 0 0
+   1       5 ram5 0 0 0 0 0 0 0 0 0 0 0
+   1       6 ram6 0 0 0 0 0 0 0 0 0 0 0
+   1       7 ram7 0 0 0 0 0 0 0 0 0 0 0
+   1       8 ram8 0 0 0 0 0 0 0 0 0 0 0
+   1       9 ram9 0 0 0 0 0 0 0 0 0 0 0
+   1      10 ram10 0 0 0 0 0 0 0 0 0 0 0
+   1      11 ram11 0 0 0 0 0 0 0 0 0 0 0
+   1      12 ram12 0 0 0 0 0 0 0 0 0 0 0
+   1      13 ram13 0 0 0 0 0 0 0 0 0 0 0
+   1      14 ram14 0 0 0 0 0 0 0 0 0 0 0
+   1      15 ram15 0 0 0 0 0 0 0 0 0 0 0
+   7       0 loop0 0 0 0 0 0 0 0 0 0 0 0
+   7       1 loop1 0 0 0 0 0 0 0 0 0 0 0
+   7       2 loop2 0 0 0 0 0 0 0 0 0 0 0
+   7       3 loop3 0 0 0 0 0 0 0 0 0 0 0
+   7       4 loop4 0 0 0 0 0 0 0 0 0 0 0
+   7       5 loop5 0 0 0 0 0 0 0 0 0 0 0
+   7       6 loop6 0 0 0 0 0 0 0 0 0 0 0
+   7       7 loop7 0 0 0 0 0 0 0 0 0 0 0
+   8       0 sda 7089 1514 243714 6329 86342 820181 7159936 1397746 0 50380 1404142
+   8       1 sda1 560 243 12192 133 64 5182 10504 955 0 218 1085
+   8       2 sda2 6375 1271 230290 6161 77130 814999 7149432 1395024 0 48563 1401264
+ 253       0 dm-0 7255 0 226842 8638 895753 0 7149432 40185506 0 50361 40194137
+ 253       1 dm-1 307 0 2456 116 0 0 0 0 0 116 116
```

Comment:
This will work in 2.6 onwards, including the 3.1x line. Confirmed with CentOS 6 and 7. If we want to support 2.4 it requires significant extra work, these metrics moved completely.